### ISSUE TITLE
NEED HELP -> What is "S: a0 BAD Invalid literal size"?

### ISSUE THREAD

**Issue by xamantra** [Enough-Software/enough_mail#158]
Here's my whole code

```dart
import 'package:enough_mail/enough_mail.dart';

const userName = 'myname@companymail.com'; // just a sample
const password = '({hKu8qM?BSx'; // just a sample but the real password looks like that
const imapServerHost = 'mail.company.com'; // just a sample
const imapServerPort = 993; // real port

void main() {
  imapExample();
}

/// Low level IMAP API usage example
Future<void> imapExample() async {
  final client = ImapClient(isLogEnabled: true);
  try {
    await client.connectToServer(imapServerHost, imapServerPort, isSecure: true);
    await client.login(userName, password);
    print('client.listMailboxes() ...');
    final mailboxes = await client.listMailboxes();
    print('mailboxes: $mailboxes');
    await client.selectInbox();
    // fetch 10 most recent messages:
    final fetchResult = await client.fetchRecentMessages(messageCount: 10, criteria: 'BODY.PEEK[]');
    for (final message in fetchResult.messages) {
      print(message);
    }
    await client.logout();
  } catch (e, trace) {
    print(['IMAP failed with $e', trace.toString()]);
  }
}
```

Here's the full logs
```bash
A: connecting to server mail.company.com:993 - secure: true
S: * OK [CAPABILITY IMAP4rev1 SASL-IR LOGIN-REFERRALS ID ENABLE IDLE NAMESPACE LITERAL+ AUTH=PLAIN AUTH=LOGIN] Dovecot ready.

C: a0 LOGIN myname@company.com (password scrambled)
S: a0 BAD Invalid literal size

[IMAP failed with null, ]
S: * BYE Disconnected for inactivity.

C: untagged not handled: [* BYE Disconnected for inactivity.] by task null
A: Done, connection closed
```

I see in the logs that it is able to connect to our company's mail server but not able to login with that `BAD Invalid literal size` error.

I'm using the latest version **1.3.0**

Do you have an idea what is `BAD Invalid literal size` error? What part of the server should I look into for this?

@robert-virkus

**Comment by xamantra** [comment]
Additional info.
I'm able to connect and login with POP client. Using the port `995`.
Same host domain, email_address and password.

**Comment by robert-virkus** [comment]
Thanks for the report and the details.
This seems to be do to the password containing the squiggly parentheses character`{`. It seems that this character must be escaped or something. Can you sign in when you quote the password, e.g. 
```dart
final quotedPassword = '"$password"'; 
await imapClient.login(userName, quotedPassword);
```

**Comment by xamantra** [comment]
Oh. I will try this.

**Comment by xamantra** [comment]
**Holy moly that worked!!**

Tried with both secure and unsecured login.

BIG THANKS @robert-virkus

**Comment by robert-virkus** [comment]
ok, cool. Please do not do this permanently, I will fix this with the next release.

**Comment by robert-virkus** [comment]
released the fix now in enough_mail v1.3.1
